### ISSUE TITLE
Support for COMMAND_CLASS_ENTRY_CONTROL

### ISSUE THREAD

**Issue by robertraaijmakers** [athombv/node-homey-zwavedriver#16]
I couldn't find any support for the COMMAND_CLASS_ENTRY_CONTROL. This class is for instance used in the BeNext Tag Reader 500. I'm trying to implement this device with advanced control (they call it "Gateway Mode"). But therefore I need the COMMAND_CLASS_ENTRY_CONTROL to be supported. If this class is implemented the Tag Reader 500 can:
- Control the UserIds belonging to the tags (this makes it possible to control the userids by Homey and then use multiple tag readers with the same tags).
- Toggle scenes (flows) with the tag reader by pressing 1,2,3 enter or other 'scene numbers' on the tag reader.
- Make Homey control the alarm (when able to 'arm' the home and when not) and give this as feedback to the Tag Reader by letting the light blink green or red.

It would be cool to implement this class as many Homey users have this device, and probably also other devices use this class for advanced control of security systems.

Tag Reader manual: http://www.benext.eu/static/manual/tagreader500.pdf (page 3 'Gateway Mode')
COMMAND_CLASS_ENTRY_CONTROL: http://z-wave.sigmadesigns.com/wp-content/uploads/2016/08/SDS12657-12-Z-Wave-Command-Class-Specification-A-M.pdf (page 244).

**Comment by RobinBol** [comment]
Why do you think this class is not supported? Can you show me some logging output directly after pairing the device?

**Comment by robertraaijmakers** [comment]
The list of command classes after pairing the device.
Also when I add the COMMAND_CLASS_ENTRY_CONTROL as a custom capability it gives an error that it isn't recognized.
-------------------------------------------------
eu.benext running...
[debug] ------------------------------------------
[debug] Node: baaec7f5-c7aa-44a3-8691-4f59e4433818
[debug] - Battery: true
[debug] - CommandClass: COMMAND_CLASS_ZWAVEPLUS_INFO
[debug] -- Version: 2
[debug] -- Commands:
[debug] --- ZWAVEPLUS_INFO_GET
[debug] --- ZWAVEPLUS_INFO_REPORT
[debug] - CommandClass: COMMAND_CLASS_MANUFACTURER_SPECIFIC
[debug] -- Version: 2
[debug] -- Commands:
[debug] --- MANUFACTURER_SPECIFIC_GET
[debug] --- MANUFACTURER_SPECIFIC_REPORT
[debug] --- DEVICE_SPECIFIC_GET
[debug] --- DEVICE_SPECIFIC_REPORT
[debug] - CommandClass: COMMAND_CLASS_VERSION
[debug] -- Version: 2
[debug] -- Commands:
[debug] --- VERSION_COMMAND_CLASS_GET
[debug] --- VERSION_COMMAND_CLASS_REPORT
[debug] --- VERSION_GET
[debug] --- VERSION_REPORT
[debug] - CommandClass: COMMAND_CLASS_DEVICE_RESET_LOCALLY
[debug] -- Version: 1
[debug] -- Commands:
[debug] --- DEVICE_RESET_LOCALLY_NOTIFICATION
[debug] - CommandClass: COMMAND_CLASS_ASSOCIATION
[debug] -- Version: 2
[debug] -- Commands:
[debug] --- ASSOCIATION_GET
[debug] --- ASSOCIATION_GROUPINGS_GET
[debug] --- ASSOCIATION_GROUPINGS_REPORT
[debug] --- ASSOCIATION_REMOVE
[debug] --- ASSOCIATION_REPORT
[debug] --- ASSOCIATION_SET
[debug] --- ASSOCIATION_SPECIFIC_GROUP_GET
[debug] --- ASSOCIATION_SPECIFIC_GROUP_REPORT
[debug] - CommandClass: COMMAND_CLASS_ASSOCIATION_GRP_INFO
[debug] -- Version: 1
[debug] -- Commands:
[debug] --- ASSOCIATION_GROUP_NAME_GET
[debug] --- ASSOCIATION_GROUP_NAME_REPORT
[debug] --- ASSOCIATION_GROUP_INFO_GET
[debug] --- ASSOCIATION_GROUP_INFO_REPORT
[debug] --- ASSOCIATION_GROUP_COMMAND_LIST_GET
[debug] --- ASSOCIATION_GROUP_COMMAND_LIST_REPORT
[debug] - CommandClass: COMMAND_CLASS_WAKE_UP
[debug] -- Version: 2
[debug] -- Commands:
[debug] --- WAKE_UP_INTERVAL_CAPABILITIES_GET
[debug] --- WAKE_UP_INTERVAL_CAPABILITIES_REPORT
[debug] --- WAKE_UP_INTERVAL_GET
[debug] --- WAKE_UP_INTERVAL_REPORT
[debug] --- WAKE_UP_INTERVAL_SET
[debug] --- WAKE_UP_NO_MORE_INFORMATION
[debug] --- WAKE_UP_NOTIFICATION
[debug] - CommandClass: COMMAND_CLASS_CONFIGURATION
[debug] -- Version: 1
[debug] -- Commands:
[debug] --- CONFIGURATION_GET
[debug] --- CONFIGURATION_REPORT
[debug] --- CONFIGURATION_SET
[debug] - CommandClass: COMMAND_CLASS_SWITCH_BINARY
[debug] -- Version: 1
[debug] -- Commands:
[debug] --- SWITCH_BINARY_GET
[debug] --- SWITCH_BINARY_REPORT
[debug] --- SWITCH_BINARY_SET
[debug] - CommandClass: COMMAND_CLASS_BATTERY
[debug] -- Version: 1
[debug] -- Commands:
[debug] --- BATTERY_GET
[debug] --- BATTERY_REPORT
[debug] - CommandClass: COMMAND_CLASS_POWERLEVEL
[debug] -- Version: 1
[debug] -- Commands:
[debug] --- POWERLEVEL_GET
[debug] --- POWERLEVEL_REPORT
[debug] --- POWERLEVEL_SET
[debug] --- POWERLEVEL_TEST_NODE_GET
[debug] --- POWERLEVEL_TEST_NODE_REPORT
[debug] --- POWERLEVEL_TEST_NODE_SET
[debug] - CommandClass: COMMAND_CLASS_FIRMWARE_UPDATE_MD
[debug] -- Version: 2
[debug] -- Commands:
[debug] --- FIRMWARE_MD_GET
[debug] --- FIRMWARE_MD_REPORT
[debug] --- FIRMWARE_UPDATE_MD_GET
[debug] --- FIRMWARE_UPDATE_MD_REPORT
[debug] --- FIRMWARE_UPDATE_MD_REQUEST_GET
[debug] --- FIRMWARE_UPDATE_MD_REQUEST_REPORT
[debug] --- FIRMWARE_UPDATE_MD_STATUS_REPORT
[debug] - CommandClass: COMMAND_CLASS_SECURITY
[debug] -- Version: 1
[debug] -- Commands:
[debug] --- NETWORK_KEY_SET
[debug] --- NETWORK_KEY_VERIFY
[debug] --- SECURITY_COMMANDS_SUPPORTED_GET
[debug] --- SECURITY_COMMANDS_SUPPORTED_REPORT
[debug] --- SECURITY_MESSAGE_ENCAPSULATION
[debug] --- SECURITY_MESSAGE_ENCAPSULATION_NONCE_GET
[debug] --- SECURITY_NONCE_GET
[debug] --- SECURITY_NONCE_REPORT
[debug] --- SECURITY_SCHEME_GET
[debug] --- SECURITY_SCHEME_INHERIT
[debug] --- SECURITY_SCHEME_REPORT
[debug] - CommandClass: COMMAND_CLASS_BASIC
[debug] -- Version: 1
[debug] -- Commands:
[debug] --- BASIC_GET
[debug] --- BASIC_REPORT
[debug] --- BASIC_SET
[debug] - CommandClass: COMMAND_CLASS_ALARM
[debug] -- Version: 2
[debug] -- Commands:
[debug] --- ALARM_GET
[debug] --- ALARM_REPORT
[debug] --- ALARM_SET
[debug] --- ALARM_TYPE_SUPPORTED_GET
[debug] --- ALARM_TYPE_SUPPORTED_REPORT
[debug] - CommandClass: COMMAND_CLASS_USER_CODE
[debug] -- Version: 1
[debug] -- Commands:
[debug] --- USER_CODE_GET
[debug] --- USER_CODE_REPORT
[debug] --- USER_CODE_SET
[debug] --- USERS_NUMBER_GET
[debug] --- USERS_NUMBER_REPORT
[debug] - CommandClass: Error
[debug] -- Version: 1
[debug] -- Commands:
[debug] - CommandClass: COMMAND_CLASS_INDICATOR
[debug] -- Version: 2
[debug] -- Commands:
[debug] --- INDICATOR_GET
[debug] --- INDICATOR_REPORT
[debug] --- INDICATOR_SET
[debug] ------------------------------------------

**Comment by WeeJeWel** [comment]
It seems the device doesn't advertise it supports the Command Class. Maybe you need to set a configuration parameter, or pair it securely?

**Comment by robertraaijmakers** [comment]
Thanks for the suggestions. I did some research and trying to see if I could tweak the settings, but I'm stuck (the device is added as a secure device). I can't change the device settings as the settings button is grayed out (and whatever I change in the settings it stays that way). 

Also there is an " Error " reported for the " Device Class Specific ". Maybe this has something to do with the grayed out button, but don't expect so myself.

I also noticed that the
`[debug] - CommandClass: Error
[debug] -- Version: 1
[debug] -- Commands:`

changed to 
`[debug] - CommandClass: undefined
[debug] -- Version: 1
[debug] -- Commands:`

I'm a bit lost... :(

I will try this weekend to see what happens when I add COMMAND_CLASS_ENTRY_CONTROL again to the app.json and the driver and if I reset my device and push the 'gateway mode' setting directly on pairing.

<img width="523" alt="tag reader 500 settings" src="https://cloud.githubusercontent.com/assets/1773099/21506641/fa307ac0-cc71-11e6-8abd-6013448f457c.PNG">

**Comment by robertraaijmakers** [comment]
Had some time today to check this already and experiment a little bit further. So, when not adding the COMMAND_CLASS_ENTRY_CONTROL as a custom capability to my device I can now edit the settings (I updated previous post to remove this 'dependency' to prevent some confusion).

But the error below is, initially, why I created the issue. When trying to add it it tells me that it is an invalid command class, this makes me assume that the command class isn't supported by Homey.

<img width="340" alt="benext invalid command class" src="https://cloud.githubusercontent.com/assets/1773099/21519292/39db6cd2-ccec-11e6-8dab-e7e92e1b915c.PNG">
